File: QsWebSoft/Xt_Popwin/W_Hssh_Tsl_Import.win.js
///<reference path="/Scripts/ExtPB_Net.js" />
///<reference path="/Scripts/MyExtPB.js" />
///<reference path="/Beta3/ExtPB_Demo.js" />

function W_Hssh_Tsl_Import() {

	 var self=this;
	 var requestor= new PBWindow();
	 var dw_1= new DataWindow( requestor );
	 var tb_1= new ToolStrip( requestor );
	 var ds_1= new DataStore( requestor );

	 if(typeof(this.bind)=="function") var baseBind=this.bind;
	 this.bind=function(win){
	 if(typeof( baseBind )=="function") baseBind(win);

		 if(requestor == null ) requestor= new PBWindow();
		 requestor.Attach(win);
		 if(dw_1 == null ) dw_1= new DataWindow( requestor );
		 dw_1.Attach( win.Control( "dw_1") ); 
		 if(tb_1 == null ) tb_1= new ToolStrip( requestor );
		 tb_1.Attach( win.Control( "tb_1") ); 
		 if(ds_1 == null ) ds_1= new DataStore( requestor );
		 ds_1.Attach( win.Control( "ds_1") ); 

		 if(typeof(self.PostOpen)=="function") 
			 win.AttachEvent("window","PostOpen",self.PostOpen); 

		 if(typeof(self.OK)=="function") 
			 win.AttachEvent("btn_1","Clicked",self.OK); 

	 }

	 if(typeof(this.unbind)=="function") var baseUnBind=this.unbind; 
	 this.unbind=function() {
	 if(typeof(this.Release)=="function") this.Release(); 
	 if(typeof(baseUnBind)=="function") baseUnBind(); 

		 dw_1.Detach(); dw_1=null; 
		 tb_1.Detach(); tb_1=null; 
		 ds_1.Detach(); ds_1=null; 
		 requestor.Detach(); requestor=null; 

		 setTimeout(CollectGarbage, 100); 
	 }

	 //以上代码由设计器自动生成,请不要手工修改
	 //ExtPB.Net JavaScript End  设计器查找标志,请不要删除或修改

    var dw_mx = null;

    this.SetDataWindow = function (dw) {
        dw_mx = dw;
        for (row = dw_mx.RowCount(); row > 0; row--) {
            dw_mx.DeleteRow(row);
        }
    }
    var column = null;
    this.SetData = function (parm) {
        column = parm;
    }
    
    //#region 窗口的OPEN事件
    this.PostOpen = function () {
//        self.OK();
        //        alert(dw_mx.RowCount());
    }
    this.OK = function () {
        if (dw_1.RowCount() == 0) {
            alert("未查到对应的信息!")
            requestor.Close();
            return;
        }
//        for (row = dw_1.RowCount(); row > 0; row--) {


//            sxrq = dw_1.GetItemString(row, "vesselname");
//            jsrq = dw_1.GetItemString(row, "voyage");
//            if (sxrq != null && sxrq != "") {
//                name = dw_1.GetItemString(row, "stepname");
//                unit = dw_1.GetItemString(row, "remark");
//                dwcode = dw_1.GetItemString(row, "ctnno");
//                zssl_Set = dw_1.GetItemString(row, "billno");
//                tsl = dw_1.GetItemString(row, "procresul");
//                hgbm = dw_1.GetItemString(row, "hgbm");
//            }
//            insertrow = dw_mx.InsertRow(0);
//            dw_mx.SetItem(insertrow, "vesselname", sxrq);
//            dw_mx.SetItem(insertrow, "voyage", jsrq);
//            dw_mx.SetItem(insertrow, "stepname", name);
//            dw_mx.SetItem(insertrow, "remark", unit);
//            dw_mx.SetItem(insertrow, "ctnno", dwcode);
//            dw_mx.SetItem(insertrow, "procresul", tsl);
//            dw_mx.SetItem(insertrow, "billno", zssl_Set);
//            dw_mx.SetItem(insertrow, "hgbm", hgbm);
//        }
        requestor.Close();
    }

    this.Exit = function () {
        requestor.Close();
    }

    this.sle_EditChanged = function (text) {
        dw_1.SetRedraw(false);
        if (text == "") {
            dw_1.SetFilter("");
            dw_1.Filter();
        }
        else {
            dw_1.SetFilter("( Upper(hs_code) like '%" + text + "%') or ( Lower(cname) like '%" + text + "%') or ( cname like '%" + text + "%')");
            dw_1.Filter();
            dw_1.Sort();
        }
        dw_1.SetRedraw(true);
    }


 }

File: QsWebSoft/Xt_Popwin/W_Hssh_Tsl_Import.win.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Text;
using System.Web;
using System.Data;
using System.Data.SqlClient;
using Sybase.DataWindow;
using TXSoft.Common;
using TXSoft.ExtPB;
using TXSoft.DataStore;
using System.Linq;
using System.IO;
using Microsoft.Win32;
using System.Configuration;
using System.Diagnostics;
using QsWebSoft.Helper;

namespace QsWebSoft.Xt_Popwin
{
    public partial class W_Hssh_Tsl_Import
    {
        public W_Hssh_Tsl_Import()
        {

        }
        public override void OnLoad()
        {
            base.OnLoad();
            ReportService report = (ReportService)dw_1.Services.Add(ServiceName.Report);
            report.RequestorDrawTitle = false;
            this.Visible = false;
            //dw_1.Retrieve();
            //dw_1.Modify("DataWindow.Readonly=yes");
            var method = "getTaxRate";
            //var code = "5407520091";
            var code = this.Request["code"].ToString(); 
            //var tsflag = this.Request["tsflag"].ToString(); 

            string strFile = AppDomain.CurrentDomain.BaseDirectory;
            var timestamp = Context.Timestamp.ToString("yyyy-MM-dd HH:mm:ss");
            string sign = WS.MD5("hd01" + "750a8953a27a4d0aa6b4a70576574ead" + timestamp);
            //method = "getVisualLogistics";
            string format = "xml";
            string uid = "hd01";
            string url = "";
            //取随机数
            Random rnd = new Random();
            int n = rnd.Next(10000, 99999);
            url = "http://api.trainer.nbeport.com/router/rest?method=" + method + "&format=" + format + "&code=" + code + "&user_id=" + uid + "&timestamp=" + timestamp + "&sign=" + sign + "&t=" + n;
            //url = Server.UrlEncode(url);
            //WriteTextLog(url);
            strFile = strFile + "XML\\" + code + ".xml";

            //WS.DownLoadAndCreateXML(url, strFile);
            var Stream = WS.WriteCardToStream(url);
            //STREAM转换为string
            StreamReader reader = new StreamReader(Stream);
            string text = reader.ReadToEnd();
           
            var row = 0;
            var sxrq = "";
            var jsrq = "";
            var insertrow = 0;
            var name = "";
            var unit = "";
            var dwcode = "";
            var zssl_Set = "";
            var tsl = "";
            var hgbm = "";
            //for (row = dw_1.RowCount; row > 0; row--)
            //{
            //    dw_1.DeleteRow(row);
            //}

            //for (row = ds_1.RowCount; row > 0; row--)
            //{
            //    ds_1.DeleteRow(row);
            //}
            ds_1.ImportString(text, Sybase.DataWindow.FileSaveAsType.Xml);
            if (ds_1.RowCount > 1)
            {
                //更新行号
                for (row = ds_1.RowCount; row > 0; row--)
                {
                    insertrow = dw_1.InsertRow(0);
                    sxrq = ds_1.GetItemString(row, "vesselname");
                    jsrq = ds_1.GetItemString(row, "voyage");
                    name = ds_1.GetItemString(row, "hgbm");
                    if (sxrq != null && sxrq != "")
                    {
                        hgbm = ds_1.GetItemString(row + 1, "stepname");
                        unit = ds_1.GetItemString(row + 1, "remark");
                        dwcode = ds_1.GetItemString(row + 1, "ctnno");
                        zssl_Set = ds_1.GetItemString(row + 1, "billno");
                        tsl = ds_1.GetItemString(row + 1, "procresul");
                    }

                    dw_1.SetItemString(insertrow, "vesselname", sxrq);
                    dw_1.SetItemString(insertrow, "voyage", jsrq);
                    dw_1.SetItemString(insertrow, "stepname", name);
                    dw_1.SetItemString(insertrow, "remark", unit);
                    dw_1.SetItemString(insertrow, "ctnno", dwcode);
                    dw_1.SetItemString(insertrow, "procresul", tsl);
                    dw_1.SetItemString(insertrow, "billno", zssl_Set);
                    dw_1.SetItemString(insertrow, "hgbm", hgbm);
                }
                for (row = dw_1.RowCount; row > 0; row--)
                {
                    if (dw_1.GetItemString(row, "stepname") == "" || dw_1.GetItemString(row, "stepname") == null)
                    {
                        dw_1.DeleteRow(row);
                    }
                }

            }
            //dw_1.ImportString(text, Sybase.DataWindow.FileSaveAsType.Xml);
            //dw_1.Retrieve();
            dw_1.Modify("DataWindow.Readonly=yes");
        }
    }
}
